lib/pipeline/RequestHandler.js

```diff
@@ -73,24 +76,29 @@ class RequestHandler extends Duplex {
     switch(request.method) {
       case 'PING':
         this._queueResponse(request, { request_id: request.id });
-        return callback();
+        break;
       case 'TAKE':
-        return this.db.take(request, this._commonReponseHandler(request, callback));
+        this.db.take(request, this._commonReponseHandler(request));
+        break;
       case 'WAIT':
-        return this.db.wait(request, this._commonReponseHandler(request, callback));
+        this.db.wait(request, this._commonReponseHandler(request));
+        break;
       case 'PUT':
-        return this.db.put(request, this._commonReponseHandler(request, callback));
+        this.db.put(request, this._commonReponseHandler(request));
+        break;
       case 'GET':
-        return this.db.get(request, this._commonReponseHandler(request, callback));
+        this.db.get(request, this._commonReponseHandler(request));
+        break;
       case 'STATUS':
-        return this.db.status({type: request.type, prefix: request.key}, (err, result) => {
+        this.db.status({type: request.type, prefix: request.key}, (err, result) => {
           if (err) { return this.emit('error', err); }
           this._queueResponse(request, result);
-          callback();
         });
+        break;
       default:
         return callback(new Error(`unknown method ${request.method}`));
     }
+    callback();
```

Comment:
Because of line 93 at this point the result of 'STATUS' will not be queue yet, are you sure that's fine?